### ISSUE TITLE
1.21 with Minestom

### ISSUE THREAD

**Issue by fucksophie** [minekube/gate#357]
Minestom and 1.21 don't seem to go well together, getting these errors when connecting.

Debug: https://pastes.dev/3miOlzjgWj
Server: https://git.sad.ovh/sophie/lobby
Client: Minecraft 1.21, vanilla

**Comment by fucksophie** [comment]
https://pastes.dev/z85zT0q4Up

**Comment by NixNux123** [comment]
Well, I have tried to reproduce your problem and it works fine. I used the latest gate version, your gate config and your minestorm server. 
In your debug log it says that it can't decode a disconnect packet because of some stuff with nbt data. Is there a Gate plugin or something that kicks the player with a custom message? Because I'm not getting this error and I have no idea where it's coming from.

![img-2024-06-22-15-21-13](https://github.com/minekube/gate/assets/95556010/84a4e137-a1cc-4faa-9b63-8de868a3ec0d)
